rpc/rpccore/submitted_transaction_cache_test.go

```diff
@@ -0,0 +1,110 @@
+package rpccore_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/rpc/rpccore"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAdd(t *testing.T) {
+	capacity := 5
+	entryTTL := time.Second
+
+	t.Run("Insert to non-full cache", func(t *testing.T) {
+		cache := rpccore.NewSubmittedTransactionsCache(capacity, entryTTL)
+		txnHash := felt.One
+		cache.Add(txnHash)
+		require.True(t, cache.Contains(txnHash))
+	})
+
+	t.Run("Inserting to full cache with expired txs", func(t *testing.T) {
+		cache := rpccore.NewSubmittedTransactionsCache(capacity, entryTTL)
+		for i := range capacity {
+			txnHash := new(felt.Felt).SetUint64(uint64(i))
+			cache.Add(*txnHash)
+		}
+		/// Expire all txs in cache.
+		time.Sleep(entryTTL)
+
+		txnHash := new(felt.Felt).SetUint64(uint64(capacity))
+		cache.Add(*txnHash)
+		require.True(t, cache.Contains(*txnHash))
+
+		for i := range capacity {
+			require.False(t, cache.Contains(*new(felt.Felt).SetUint64(uint64(i))))
+		}
+	})
+
+	t.Run("Inserting to full cache with non-expired txs", func(t *testing.T) {
+		cache := rpccore.NewSubmittedTransactionsCache(capacity, entryTTL)
+		txnHash := felt.Zero
+		cache.Add(txnHash)
+		require.True(t, cache.Contains(felt.Zero))
+		// Populates the cache to cap.
+		for i := 1; i < capacity; i++ {
+			txnHash := new(felt.Felt).SetUint64(uint64(i))
+			cache.Add(*txnHash)
+		}
+
+		txnHash = *new(felt.Felt).SetUint64(uint64(capacity))
+		// Expected to remove the least-recently-used entry.
+		cache.Add(txnHash)
+		require.True(t, cache.Contains(txnHash))
+		require.False(t, cache.Contains(felt.Zero))
+	})
```

Comment:
nit: readability regarding felt.Zero
```suggestion
	t.Run("Inserting to full cache with non-expired txs", func(t *testing.T) {
		cache := rpccore.NewSubmittedTransactionsCache(capacity, entryTTL)
		firstTxnHash := felt.Zero
		cache.Add(firstTxnHash)
		require.True(t, cache.Contains(firstTxnHash))
		// Populates the cache to cap.
		for i := 1; i < capacity; i++ {
			txnHash := new(felt.Felt).SetUint64(uint64(i))
			cache.Add(*txnHash)
		}

		txnHash := *new(felt.Felt).SetUint64(uint64(capacity))
		// Expected to remove the least-recently-used entry.
		cache.Add(txnHash)
		require.True(t, cache.Contains(txnHash))
		require.False(t, cache.Contains(firstTxnHash))
	})
```